src/app/upgrade.cpp

```diff
@@ -82,3 +82,36 @@ bool upgrade(const bool /*ask*/)
     exportWebUIHttpsFiles();
     return true;
 }
+
+void handleChangedDefaults(const DefaultPreferencesMode mode)
+{
+    struct DefaultValue
+    {
+        QVariant legacy;
+        QVariant current;
+
+        QVariant operator[](const DefaultPreferencesMode mode) const
+        {
+            switch (mode)
+            {
+            case DefaultPreferencesMode::Legacy:
+                return legacy;
+            case DefaultPreferencesMode::Current:
+                return current;
+            default:
+                Q_ASSERT(false);
+                return current;
+            }
+        }
+    };
+
+    const QHash<QString, DefaultValue> changedDefaults {
+        {QLatin1String {"BitTorrent/Session/QueueingSystemEnabled"}, {true, false}}
```

Comment:
You might consider adding this here.
{QLatin1String {"BitTorrent/Session/UseRandomPort"}, {false, true}}
Would close #11935